.github/workflows/ci.yml

```diff
@@ -110,3 +112,38 @@ jobs:
         timeout_minutes: 15
         max_attempts: 2
         command: ./tools/test_monitor/run-tests.sh
+  flow-legacy-client-compatibility-test:
+    name: Client Compatibility Tests With Observer
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version:
+          - 1.16
+          - 1.17
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ matrix.go-version}}
+    - name: Checkout repo
+      uses: actions/checkout@v2
+    - name: Build relic
+      run: make crypto_setup_gopath
+    - name: Docker build
+      run: make install-tools
+    - name: Install Flow Client
+      run: sh -ci "$(curl -fsSL https://storage.googleapis.com/flow-cli/install.sh)" && mv ~/.local/bin/flow /tmp
+    - name: Set up Localnet
+      run: bash -c 'cd integration/localnet/ && make -e OBSERVER=1 init && make start'
+    - name: Ensure Observer is started
+      run: docker ps -q -f name=localnet_observer_1_1
+    - name: Get Version ensuring the official client build can connect
+      run: /tmp/flow -f integration/localnet/flow-localnet.json -n observer version
+    - name: Get Status ensuring the official client build can connect
+      run: /tmp/flow -f integration/localnet/flow-localnet.json -n observer status
+    - name: Wait for finalized blocks and check them with Observer
+      # || true = non-breaking due to rpc change. We do access and observer together to fetch the same block
+      run: /tmp/flow -f integration/localnet/flow-localnet.json -n access blocks get latest && (/tmp/flow -f integration/localnet/flow-localnet.json -n observer blocks get latest || true)
```

Comment:
I don't think this is actually testing anything. Right now it calls these endpoints, then ignores the result and the response code. If we're not checking that the response is sane/correct, what's the purpose of this suite? 

At a minimum, we should check that we got a successful response back. Better if we compare that the AN and observer are in sync by getting the block ID from the AN, then querying the observer for that block. Best would be some formal integration test that verifies the data within the block responses are the same as mentioned in a previous comment.

Reply:
Yes, we actually have such a test, and we are working on adding it as another pull request. (hopefully early next week)

Let's discuss the timing in a private chat.

Reply:
"At a minimum, we should check that we got a successful response back."
I think these two items test what you really expect, don't they?
```
    - name: Get Version ensuring the official client build can connect
      run: ~/.local/bin/flow -f integration/localnet/flow-localnet.json -n observer version
    - name: Get Status ensuring the official client build can connect
      run: ~/.local/bin/flow -f integration/localnet/flow-localnet.json -n observer status
```

Reply:
`flow ... version` just returns the flow-cli client version. I don't believe it actually makes a request to the AN.

`flow ... status` does make a request, but it doesn't return a non-zero response code if the network is offline, so you have the check the contents of the response. e.g.
```
flow -n localnet status -o json | jq '.status' | grep "ONLINE"
```

It's fine if the more robust checks are added as unit/integration tests in a separate PR.

Reply:
Thank you, Peter. It makes sense. I will add this to a later PR.

Reply:
I added the requested changes here https://github.com/GetElastech/flow-go/commit/6f60ededac4a24e3182b9a51bcb4f5238620054f